### PR TITLE
[integrations] Improve markdown rendering

### PR DESCRIPTION
**Pull request by thiskevinwang**
https://dev-portal-git-kw-improve-markdown-rendering-hashicorp.vercel.app

## 🔗 Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](url) 🎟️

## 🤷 Why

Consistent look/feel of code blocks and markdown in general

## 🛠️ How

This leverages `next-mdx-remote` and our typical suite of remark/rehype plugins and client-side react components

## 📸 Design Screenshots

CleanShot 2022-12-15 at 16 33 54@2x

## 💭 Anything else?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203548351777713

### CODE REVIEW

File: src/views/product-integration/index.tsx [hunk 1]
```diff
@@ -11,14 +11,15 @@ import { Integration } from 'lib/integrations-api-client/integration'
 import { Release, ReleaseComponent } from 'lib/integrations-api-client/release'
 import { useRouter } from 'next/router'
 import { useMemo } from 'react'
-import ReactMarkdown from 'react-markdown'
```
Comment (BrandonRomano) [inline src/views/product-integration/index.tsx:14]: @thiskevinwang I think we can remove this from dependencies too.

This should be the only place we were using this & I added `react-markdown` in our integrations PR myself (wasn't already in DevDot).

File: src/views/product-integration/components/component-tab-content/index.tsx [hunk 2]
```diff
@@ -4,21 +4,28 @@ import {
...
+	const readmeContent = (
+		<MDXRemote
+			{...component.readmeMdxSource}
+			components={defaultMdxComponents({})}
+		/>
+	)
 	return (
 		<div className={s.componentTabContent}>
-			{component.readme && <ReactMarkdown>{component.readme}</ReactMarkdown>}
+			{component.readmeMdxSource ? readmeContent : null}
```
Comment (BrandonRomano) [inline src/views/product-integration/components/component-tab-content/index.tsx:28]: There appears to be a margin-top here (I believe coming from the H2).

Is there something we can do about this (e.g. first child no margin top?)

CleanShot 2022-12-19 at 10 12 48@2x

**Review by BrandonRomano — CHANGES_REQUESTED** [review]
- [ ] Can remove margin top on first element
- [ ] Let's remove the react-markdown dependency

File: src/views/product-integration/index.tsx (hunk 1 above)
Comment (thiskevinwang) [reply src/views/product-integration/index.tsx:14]: `react-markdown` is still used by `VariableGroupList` currently. My thought was to leave it for now, at least until Variable Groups' README content are processed by our `next-mdx-remote` pipeline as well.

It seemed like high-effort, low-reward, so I didn't do that part yet last week

Comment (BrandonRomano) [reply src/views/product-integration/index.tsx:14]: Oh!  Yes okay -- let's do that my bad!

File: src/views/product-integration/components/component-tab-content/index.tsx (hunk 2 above)
Comment (thiskevinwang) [reply src/views/product-integration/components/component-tab-content/index.tsx:28]: I'll see what I can do!

(note, it's being defined here https://github.com/hashicorp/dev-portal/blob/kw/improve-markdown-rendering/src/components/dev-dot-content/mdx-components/mdx-headings/mdx-headings.module.css#L44-L48, and applied when we opt into using `MdxH1`, `MdxH2`, etc for our markdown heading elements)

**Review by BrandonRomano — APPROVED** [review]

**Pull request merged**